golang/cosmos/e2e_test/README.md

```diff
@@ -0,0 +1,100 @@
+- `E2ETEST_CHAINIMAGE_AGORIC` - the value of this will be used specific the repository & version of docker image to use for the agoric chain. a valid value must have a semicolon and be formatted as `repository:tag`. ex: `E2ETEST_CHAINIMAGE_AGORIC="ghcr.io/agoric/agoricinterchain:latest"`
+- `E2ETEST_RELAYERNAME` - set to `"cosmos"` or `"hermes"` to choose the relayer type
+- `E2ETEST_BLOCKS_TO_WAIT` - set to a number to control how many blocks to wait for an ACK from an IBC transfer and how many blocks to wait for TX settlement.
+
+ex:
+```
+# run PFM tests with the:
+# - first chains as agoric
+# - the second chain as osmosis
+# - the third chain as stride
+# - the fourth chain as gaia
+# - use the hermes relayer
+# - note `-timeout 20m` which is included because these tests can be very slow 
+$ E2ETEST_CHAINNAME0=agoric E2ETEST_CHAINNAME1=osmosis E2ETEST_CHAINNAME2=stride E2ETEST_CHAINNAME3=gaia E2ETEST_RELAYERNAME=hermes go test -timeout 20m -v -run ^TestPacketForwardMiddleware
+```
+
+## Debugging Flakes
+
+`error in transaction (code: 6): failed to execute message; message index: 0: 809C699215C5BC041D4035479D28C30EAB11DBC5953E45D4545346B78482D82C: denomination trace not found`
+- Flakey `interchaintest` has failed to properly run commands on relayers
+
+`Wrong user balance on chain<blah> expected[<foo>>] actual[<bar>]`
+- If you see `error in transaction (code: 6): failed to execute message; message index: 0: 809C699215C5BC041D4035479D28C30EAB11DBC5953E45D4545346B78482D82C: denomination trace not found` immediately below this error then the relayer fell over.
+- If not, it's possible the test simply didn't wait enough blocks for the transactions to settle
+
+If the tests terminate abnormally they may leave Docker containers or volumes stranded. These are safe to delete.
+
+## Details on the Custom Fork of `interchaintest`
+
+The [Agoric custom fork of interchaintest](https://github.com/Agoric-labs/interchaintest) is necessary because the agoric chain does not include the `x/crisis` module. However, `interchaintest` passes `agd` unsupported command line arguments to configure `x/crisis`. The fork contains a patch to skip these command line arguments when `interchaintest.ChainSpec.ChainConfig.NoCrisisModule` is set.
+
+The exact git tag version of Agoric custom fork of interchaintest must be specified in the `go.mod` for this test suite. It is important to update that tag in the fork and in `go.mod` whenever the fork has a new update.
+
+## Building a Compatible Docker Image
+
+`interchaintest` relies on a partner tool [`heighliner`](https://github.com/strangelove-ventures/heighliner). `heighliner` modified Docker images to make them compatible with `interchaintest`. To build an Agoric compatible docker image:
+
+Get the Agoric compatible changes (one time):
+```
+$ git clone git@github.com:strangelove-ventures/heighliner.git
+$ cd heighliner
+$ git remote add toliaqat git@github.com:toliaqat/heighliner.git
+$ git pull toliaqat
```

Comment:
I got below error here

```
git pull toliaqat
Confirm user presence for key ED25519-SK SHA256:F0zHI5oyzBKE4NVE3NrWOKx66JhphjwPzJqkGyo1+dM
User presence confirmed
You asked to pull from the remote 'toliaqat', but did not specify
a branch. Because this is not the default configured remote
for your current branch, you must specify a branch on the command line.
```

Did you mean `git fetch toliaqat`